src/with-component.js

```diff
@@ -1,13 +1,18 @@
 import { h, render } from 'preact';
 
-import { HTMLElement } from './util';
+import { HTMLElement, sym } from './util';
 import { withProps } from './with-props';
 import { withRender } from './with-render';
 import { withUnique } from './with-unique';
 
+const _preactDom = sym('_preactDom');
+
 export const withComponent = (Base = HTMLElement) => class extends withUnique(withRender(withProps(Base))) {
   rendererCallback (shadowRoot, renderCallback) {
-    render(renderCallback(), shadowRoot);
+    const dom = render(renderCallback(), shadowRoot, this[_preactDom]);
+    if (!this[_preactDom]) {
```

Comment:
I always have avoided that out of perf fears. Is setting a property over-and-over again like that basically ignored when you're setting it to the same value over time?  Or is the branching caused by a conditional worse?